src/main/kotlin/net/theevilreaper/dartpoet/function/FunctionSpec.kt

```diff
@@ -42,17 +42,20 @@ class FunctionSpec(
     internal val docs = builder.docs
     internal val hasDocs = builder.docs.isNotEmpty()
 
-    private val namedParameters: Set<ParameterSpec> = if (parameters.isEmpty()) {
-        setOf()
-    } else {
-        parameters.filter { it.isNamed }.toSet()
-    }
+    internal val requiredParameters = builder.parameters.filter { it.isRequired }.toImmutableSet()
+    internal val namedParameters = builder.parameters.filter { it.isNamed }.toImmutableSet()
+
+    internal val withTrailingSpaceComma = requiredParameters.isNotEmpty() || namedParameters.isNotEmpty()
 
     init {
         //check(!isTypeDef && annotation.isNotEmpty()) { "A typedef can't have annotations" }
         require(name.trim().isNotEmpty()) { "The name of a function can't be empty" }
         require(body.isEmpty() || !modifiers.contains(DartModifier.ABSTRACT)) { "An abstract method can't have a body" }
 
+        if (namedParameters.isNotEmpty() && requiredParameters.isNotEmpty()) {
+            throw IllegalArgumentException("A function only can have named or required parameters and not both of them")
+        }
+
         if (isGetter && asSetter) {
             throw IllegalArgumentException("The function can't be a setter and a getter twice")
```

Comment:
I think "The function can't be a setter and a getter at the same time" would be more clear.